com.unity.multiplayer.mlapi/Runtime/Configuration/NetworkConfig.cs

```diff
@@ -42,22 +42,42 @@ public class NetworkConfig
-        /// A list of spawnable prefabs
+        /// The previously set default player prefab
+        /// Note: this is only for an editor specific scenario where we need to determine during the
+        /// NetworkManager.OnValidate invocation period there is no PlayerPrefab value yet there is a
+        /// NetworkPrefab entry that has IsPlayer set to true.  This is also used to determine if
+        /// we are loading from a previous MLAPI version where the NetworkPrefab.IsPlayer was predominantly
+        /// used.
+        /// There are two scenarios where this is used:
+        /// 1.) If the PreviousPlayerPrefab and PlayerPrefab values are not set but a NetworkPrefab has
+        /// an entry with the NetworkPrefab.IsPlayer set to true, then we want to assign the PlayerPrefab
+        /// and PreviousPlayerPrefab
+        /// 2.) If the PlayerPrefab is null, the PreviousPlayer prefab is not, a NetworkPrefab entry
+        /// has its NetworkPrefab.IsPlayer set to true, and that NetworkPrefab entry's prefab's
+        /// NetworkObject.GlobalObjectIdHash is the same as the PreviousPlayerPrefab's NetworkObject.GlobalObjectIdHash
+        /// then the user deleted the PlayerPrefab so we can mark the NetworkPrefab.IsPlayer value of the matching
+        /// NetworkPrefab to false and the PreviousPlayerPrefab to null.
+        /// (The next time PlayerPrefab is set see scenario #1)
         /// </summary>
-        [Tooltip("The prefabs that can be spawned across the network")]
-        public List<NetworkPrefab> NetworkPrefabs = new List<NetworkPrefab>();
+        internal GameObject PreviousPlayerPrefab;
 
         /// <summary>
         /// The default player prefab
         /// </summary>
-        [SerializeReference]
-        internal uint PlayerPrefabHash;
+        [Tooltip("When set, the network manager will automatically create and spawn the assigned player prefab.  The Player Prefab can be overriden by adding it to the NetworkPrefabs list and selecting override.")]
+        public GameObject PlayerPrefab;
+
+        /// <summary>
+        /// A list of spawnable prefabs
+        /// </summary>
+        [Tooltip("The prefabs that can be spawned across the network")]
+        public List<NetworkPrefab> NetworkPrefabs = new List<NetworkPrefab>();
 
         /// <summary>
-        /// Whether or not a player object should be created by default. This value can be overridden on a case by case basis with ConnectionApproval.
+        /// This dictionary provides a quick way to check and see if a NetworkPrefab has a NetworkPrefab override.
+        /// Generated at runtime and OnValidate
         /// </summary>
-        [Tooltip("Whether or not a player object should be created by default. This value can be overridden on a case by case basis with ConnectionApproval.")]
-        public bool CreatePlayerPrefab = true;
+        public Dictionary<uint, NetworkPrefab>NetworkPrefabOverrideLinks = new Dictionary<uint, NetworkPrefab>();
```

Comment:
2 questions:
1. Is this field will be serialized to disk as a part of NetworkConfig?
2. should it be public or internal/private?

Reply:
I think this should be internal...